cpp/arcticdb/storage/storage_factory.hpp

```diff
@@ -18,7 +18,7 @@ namespace arcticdb {
         std::shared_ptr<Storage> create_storage(
                 const LibraryPath &library_path,
                 OpenMode mode,
-                const NativeVariantStorage& storage_descriptor);
+                const s3::S3Settings& storage_descriptor);
```

Comment:
This seems wrong? This could create any type of storage, not just S3?

Reply:
I have consolidated the rest of storages setup calling below function at line 22.